pkg/build/controller/build/defaults/defaults.go

```diff
@@ -112,6 +114,17 @@ func (b BuildDefaults) applyPodDefaults(pod *corev1.Pod, isCustomBuild bool) {
 		// All env vars are allowed to be set in a custom build pod, the user already has
 		// total control over the env+logic in a custom build pod anyway.
 		externalEnv := make([]corev1.EnvVar, len(b.Config.Env))
+		if b.DefaultProxy != nil {
+			if len(b.DefaultProxy.HTTPProxy) != 0 {
```

Comment:
Don't check for empty here - we assume that if we have a `ProxySpec`, empty strings are intentional.